### PR TITLE
Generic Binding improvement

### PR DESCRIPTION
**Pull request by andystaples**
Replaced GenericBinding with two different binding types - InputBinding and OutputBinding
Also, fixed the .psd1 and test issues that arose with refactoring the module name

### CODE REVIEW

File: src/BindingExtractor.cs [hunk 1]
```diff
@@ -69,6 +69,7 @@ public static bool hasSupportedBinding(Type bindingType)
 
         public static BindingInformation? ExtractInputBinding(AttributeAst attribute, ParameterAst parameter)
         {
+            Console.WriteLine(attribute.TypeName.Name);
```
Comment (Francisco-Gamino) [inline src/BindingExtractor.cs:72]: Do we need this line?

File: src/BundledBindings/InputBinding.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            if (string.IsNullOrEmpty(bindingName))
+            {
+                bindingName = parameter.Name.VariablePath.UserPath;
+            }
+
+            bindingInformation.Direction = BindingDirection;
+
+            if (problems.Count > 0)
+            {
+                throw new Exception(AzPowerShellSdkStrings.InputBindingProblemsExist + string.Join("\n", problems));
```
Comment (Francisco-Gamino) [inline src/BundledBindings/InputBinding.cs:38]: Could you please share what the error message would look like including the `FullyQualifiedErrorId`?

File: src/resources/AzPowerShellSdkStrings.Designer.cs [hunk 3]
```diff
@@ -61,7 +61,7 @@ internal class AzPowerShellSdkStrings {
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to Could not add additional information with name {name} to binding {bindingName} because no binding with this name was found.
```
Comment (Francisco-Gamino) [inline src/resources/AzPowerShellSdkStrings.Designer.cs:64]: In the `resources` folder we should only have this file: `/resources/AzPowerShellSdkStrings.resx`. What is the other one used for?

Comment (andystaples) [reply src/resources/AzPowerShellSdkStrings.Designer.cs:64]: Just a file autocreated by Visual Studio. I removed it.

File: src/BindingExtractor.cs (hunk 1 above)
Comment (andystaples) [reply src/BindingExtractor.cs:72]: No, removed

**Review by Francisco-Gamino — APPROVED** [review]
LGTM

**Pull request merged**